crates/autopilot/src/run_loop.rs

```diff
@@ -211,20 +226,32 @@ impl RunLoop {
                 return;
             }
 
-            tracing::info!(driver = %driver.name, "settling");
-            let submission_start = Instant::now();
-            match self
-                .settle(driver, solution, auction_id, block_deadline)
-                .await
-            {
-                Ok(()) => Metrics::settle_ok(driver, submission_start.elapsed()),
-                Err(err) => {
-                    Metrics::settle_err(driver, submission_start.elapsed(), &err);
-                    tracing::warn!(?err, driver = %driver.name, "settlement failed");
+            let solution_id = solution.id();
+            let solved_order_uids = solution.orders().keys().cloned().collect();
+            let self_clone = self.clone();
+            let driver_clone = driver.clone();
+            tokio::task::spawn(async move {
+                tracing::info!(driver = %driver_clone.name, "settling");
+                let submission_start = Instant::now();
+
+                match self_clone
+                    .settle(
+                        driver_clone.clone(),
+                        solution_id,
+                        solved_order_uids,
+                        auction_id,
+                        block_deadline,
+                    )
+                    .await
```

Comment:
Before executing it, we should probably have a new lock to avoid cases when multiple settlements are sent in parallel. Does it make sense?

Reply:
Yep, makes sense. Otherwise the same solver might win 2 auctions after another which can make the nonce management a lot more complex.